app/controllers/agent/business/BusinessAccountingMethodController.scala

```diff
@@ -80,15 +78,15 @@ class BusinessAccountingMethodController @Inject()(val authService: AuthService,
       }
   }
 
-  def backUrl(isEditMode: Boolean, incomeSourceModel: IncomeSourceModel, matchTaxYear: MatchTaxYearModel): String = {
+  def backUrl(isEditMode: Boolean, incomeSourceModel: IncomeSourceModel): String = {
     if (isEditMode) {
       controllers.agent.routes.CheckYourAnswersController.show().url
     } else {
-      (matchTaxYear.matchTaxYear, incomeSourceModel) match {
-        case (No, _) => controllers.agent.business.routes.BusinessAccountingPeriodDateController.show().url
-        case (Yes, IncomeSourceModel(true, false, _)) => controllers.agent.business.routes.WhatYearToSignUpController.show().url
-        case _ => controllers.agent.business.routes.MatchTaxYearController.show().url
+      incomeSourceModel match {
+        case IncomeSourceModel(false, true, _) => controllers.agent.routes.IncomeSourceController.show().url
+        case IncomeSourceModel(true, false, _) => controllers.agent.business.routes.WhatYearToSignUpController.show().url
+        case _ => controllers.agent.routes.IncomeSourceController.show().url
```

Comment:
The cases here should be:
```suggestion
        case IncomeSourceModel(true, false, _) => controllers.agent.business.routes.WhatYearToSignUpController.show().url
        case _ => controllers.agent.routes.BusinessNameController.show().url
```
This page shouldn't go back to income sources